genai-perf/genai_perf/config/input/config_perf_analyzer.py

```diff
@@ -77,37 +76,63 @@ def parse(self, perf_analyzer: Dict[str, Any]) -> None:
                 self.stability_percentage = value
             elif key == "measurement_interval":
                 self.measurement_interval = value
-            elif key == "request_count":
-                self.request_count.parse(value)
+            elif key == "warmup_request_count":
+                self.warmup_request_count = value
+            elif key == "measurement":
+                self.measurement.parse(value)
             else:
                 raise ValueError(
                     f"User Config: {key} is not a valid perf_analyzer parameter"
                 )
 
+    ###########################################################################
+    # Infer Methods
+    ###########################################################################
+    def infer_settings(self) -> None:
+        self.measurement._infer_measurement_num_based_on_mode()
 
-class ConfigRequestCount(BaseConfig):
+
+###########################################################################
+# Sub-Config Classes
+###########################################################################
+class ConfigPerfAnalyzerMeasurement(BaseConfig):
     def __init__(self) -> None:
         super().__init__()
-        self.num: Any = ConfigField(
-            default=RequestCountDefaults.NUM,
-            bounds={"min": 0},
-            verbose_template_comment="The number of requests to use for measurement.\
-                \nBy default, the benchmark does not terminate based on request count.\
-                \nInstead, it continues until stabilization is detected.",
+
+        self.mode: Any = ConfigField(
+            default=PerfAnalyzerMeasurementDefaults.MODE,
+            choices=PerfAnalyzerMeasurementMode,
+            verbose_template_comment="The mode of measurement to use in PA: request_count or interval",
         )
-        self.warmup: Any = ConfigField(
-            default=RequestCountDefaults.WARMUP,
+        self.num: Any = ConfigField(
+            default=PerfAnalyzerMeasurementDefaults.NUM,
             bounds={"min": 0},
-            verbose_template_comment="The number of warmup requests to send before benchmarking.",
+            template_comment="If left unset, the default value will be inferred based on the mode.",
+            verbose_template_comment="The number to use for measurement. By default, this will be inferred based on the mode.",
         )
 
-    def parse(self, request_count: Dict[str, Any]) -> None:
-        for key, value in request_count.items():
-            if key == "num":
+    def parse(self, perf_analyzer_measurement: Dict[str, Any]) -> None:
+        for key, value in perf_analyzer_measurement.items():
```

Comment:
can we give more meaningful names to the variables? `key` and `value` seems very generic and didn't really help me understanding the code.